### PR TITLE
Implement comparable on EventData

### PR DESCRIPTION
**Pull request by Fokko**
## Description
I would like to implement the Comparable interface, so we can easily sort the messages based on the sequence.

This will give us the possibility to just use `.sort`, instead of implementing `.sortWith`:
https://github.com/Azure/azure-event-hubs-spark/blob/7ceae44f3bfe61575d24c94fe0749a70eeccbc11/core/src/main/scala/org/apache/spark/eventhubs/client/CachedEventHubsReceiver.scala#L137-L138

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

**Comment by Fokko** [comment]
PTAL @sjkwak

File: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/EventDataImpl.java [hunk 1]
```diff
@@ -258,4 +258,17 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
...
+    public int compareTo(EventData other) {
+        long seqThis = this.getSystemProperties().getSequenceNumber();
+        long seqOther = other.getSystemProperties().getSequenceNumber();
+
+        if(seqThis < seqOther) {
+          return -1;
+        } else if (seqThis > seqOther){
+          return 1;
+        }
+        return 0;
```
Comment (sjkwak) [inline azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/EventDataImpl.java]: You could use Long.comapre() method to simplify it - Long.compare(this.getSystemProperties().getSequenceNumber(), other.getSystemProperties().getSequenceNumber());

Comment (Fokko) [reply azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/EventDataImpl.java]: Perfect suggestion, I've updated the code.

**Comment by Fokko** [comment]
Thanks @sjkwak I've reworked the PR

**Pull request merged**